### PR TITLE
Trashing old code

### PR DESCRIPTION
**Pull request by weerd**
### What does this PR do?

This PR does some cleanup to remove a lot of code no longer in use after a few of the updates to how the signup logic works in #1184 and #1186. It also makes an update to add logging in the `CampaignSignupsController@index()` method and removes the use of `legacyCampaign` and `legacyCampaignRun` references in various components.

### What are the relevant tickets/cards?

Refs [Pivotal ID #160602918](https://www.pivotaltracker.com/story/show/160602918)

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: app/Http/Controllers/Api/CampaignSignupsController.php [hunk 1]
```diff
@@ -41,7 +41,13 @@ public function index($id, Request $request)
 
         $query['filter']['campaign_id'] = $id;
 
-        return $this->signupRepository->getSignups($query);
+        Log::info('[Phoenix] CampaignSignupsController@index request data:', $query);
```
[the remark this answers is not in the document]
Comment (weerd) [reply app/Http/Controllers/Api/CampaignSignupsController.php]: Oh, I kinda put them in on purpose, haha. Felt they'd be helpful in debugging and looking into what's being requested and associated responses.

[the remark this answers is not in the document]
Comment (DFurnes) [reply app/Http/Controllers/Api/CampaignSignupsController.php]: I feel like logging the full request and response may be a little verbose for an `info` log level (unless there's a good use case I'm missing). Want to swap these to `debug` instead?

[the remark this answers is not in the document]
Comment (weerd) [reply app/Http/Controllers/Api/CampaignSignupsController.php]: Oh sure. Yeah, that sounds good. Want me to also do that for similar logs in the `CampaignPostsController` as well?

[the remark this answers is not in the document]
Comment (DFurnes) [reply app/Http/Controllers/Api/CampaignSignupsController.php]: Ah, missed those! Yes please! 🙇

**Comment by weerd** [comment]
😂

**Pull request merged**